runtime/src/lib.rs

```diff
@@ -365,15 +418,55 @@ impl Convert<Balance, RelayChainBalance> for NativeToRelay {
     }
 }
 
+pub struct CurrencyIdConvert;
+impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+    fn convert(id: CurrencyId) -> Option<MultiLocation> {
+        Some(X2(Parent, Parachain { id: ParachainInfo::get().into() }))
+	}
+}
+impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(location: MultiLocation) -> Option<CurrencyId> {
+		match location {
+		    X2(Parent, Parachain { id }) if ParaId::from(id) == ParachainInfo::get() => Some(CurrencyId {}),
+			_ => None,
+		}
+	}
+}
+impl Convert<MultiAsset, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(asset: MultiAsset) -> Option<CurrencyId> {
+		if let MultiAsset::ConcreteFungible { id, amount: _ } = asset {
+			Self::convert(id)
+		} else {
+			None
+		}
+	}
+}
+
+pub struct HandleXcm;
+impl orml_xcm_support::XcmHandler<AccountId, Call> for HandleXcm {
+    fn execute_xcm(origin: AccountId, xcm: xcm::v0::Xcm<Call>) -> sp_runtime::DispatchResult {
+        use xcm::v0::ExecuteXcm;
+		let xcm_origin = X1(Junction::AccountId32 {
+			network: xcm::v0::NetworkId::Any,
+			id: AccountId32Convert::convert(origin),
+		});
+		XcmExecutor::<XcmConfig>::execute_xcm(xcm_origin, xcm, 1000);
```

Comment:
Need to catch the Result here and fire an event indicating success / failure

Reply:
Somehow I feel that this should be in its own pallet and make the lib hold minimum logic. WDYT?

Reply:
What does the 1000 mean in this context?